modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java

```diff
@@ -142,6 +144,14 @@ public static void send(
             headers.put("Cookie", session);
         }
 
+        if (originalInMsgCtx.isPropertyTrue(PassThroughConstants.CORRELATION_LOG_STATE_PROPERTY)) {
+           if (originalInMsgCtx.getProperty(PassThroughConstants.CORRELATION_ID) == null){
```

Comment:
Shall we fix the formatting issue here?

Reply:
fixed in https://github.com/wso2/wso2-synapse/pull/1305